substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/logging/JfrLogConfiguration.java

```
@@ -0,0 +1,161 @@
+    INSTANCE;
+
+    private boolean loggingEnabled = false;
+    private JfrLogSelection[] selections;
+
+    public boolean shouldLog(int tagSetId, int level) {
+        if (!loggingEnabled) {
+            return false;
+        }
+        Optional<LogLevel> tagSetLogLevel = JfrLogTagSet.fromTagSetId(tagSetId).getLevel();
+        // LogLevel#level is not accessible, so we have to use ordinal + 1
+        return tagSetLogLevel.isEmpty() ? false : tagSetLogLevel.get().ordinal() + 1 <= level;
+    }
+
+    public void parse(String config) {
+        if (config.isBlank()) {
+            return;
+        } else if (config.toUpperCase().equals("HELP")) {
+            printHelp();
+            System.exit(0);
+        }
+
+        loggingEnabled = true;
+
+        String[] splitConfig = config.split(",");
+        selections = new JfrLogSelection[splitConfig.length];
+
+        int index = 0;
+        for (String s : splitConfig) {
+            JfrLogSelection selection = new JfrLogSelection();
+            selection.parse(s);
+            selections[index++] = selection;
+        }
+        setLogTagSetLevels();
+        verifySelections();
+    }
+
+    private void setLogTagSetLevels() {
+        for (JfrLogTagSet tagSet : JfrLogTagSet.values()) {
+            Optional<LogLevel> level = Optional.empty();
+            for (JfrLogSelection selection : selections) {
+                if ((selection.wildcard && tagSet.getTags().containsAll(selection.tags))
+                        || (selection.tags.equals(tagSet.getTags()))) {
+                    level = Optional.of(selection.level);
+                    selection.matchesATagSet = true;
+                }
+            }
+            tagSet.setLevel(level);
+        }
+    }
+
+    private void verifySelections() {
+        for (JfrLogSelection selection : selections) {
+            if (!selection.matchesATagSet) {
+                Log.log().string("error: No tag set matches tag combination ")
```

For this kind of validation, we usually throw a `UserError` instead of printing and exiting.